presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveIgnoreAbsentPartitions.java

```diff
@@ -0,0 +1,92 @@
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive;
+
+import com.google.inject.Inject;
+import io.prestosql.tempto.ProductTest;
+import io.prestosql.tempto.Requirement;
+import io.prestosql.tempto.RequirementsProvider;
+import io.prestosql.tempto.configuration.Configuration;
+import io.prestosql.tempto.fulfillment.table.MutableTablesState;
+import io.prestosql.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.fulfillment.table.MutableTableRequirement.State.LOADED;
+import static io.prestosql.tempto.fulfillment.table.TableRequirements.mutableTable;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.hive.HiveTableDefinitions.NATION_PARTITIONED_BY_BIGINT_REGIONKEY;
+import static java.lang.String.format;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestHiveIgnoreAbsentPartitions
+        extends ProductTest
+        implements RequirementsProvider
+{
+    @Inject
+    private MutableTablesState tablesState;
+
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return mutableTable(NATION_PARTITIONED_BY_BIGINT_REGIONKEY, "test_table", LOADED);
+    }
+
+    @Test
+    public void testIgnoreAbsentPartitionsWithSessionProperties()
+    {
+        String tableNameInDatabase = tablesState.get("test_table").getNameInDatabase();
+        String partitionPath = format("/user/hive/warehouse/%s/p_regionkey=9999", tableNameInDatabase);
+
+        assertThat(query("SELECT count(*) FROM " + tableNameInDatabase)).containsOnly(row(15));
+
+        assertFalse(hdfsClient.exist(partitionPath), format("Expected partition %s to not exist", tableNameInDatabase));
+        query(format("CALL hive.system.create_empty_partition('default', '%s', array['p_regionkey'], array['9999'])", tableNameInDatabase));
+
+        query("SET SESSION hive.ignore_absent_partitions = false");
+        hdfsClient.delete(partitionPath);
+        assertFalse(hdfsClient.exist(partitionPath), format("Expected partition %s to not exist", partitionPath));
+        assertThat(() -> query("SELECT count(*) FROM " + tableNameInDatabase)).failsWithMessage("Partition location does not exist");
+
+        query("SET SESSION hive.ignore_absent_partitions = true");
+        assertThat(query("SELECT count(*) FROM " + tableNameInDatabase)).containsOnly(row(15));
+    }
+
+    @Test
+    public void testShouldThrowErrorOnUnpartitionedTableMissingData()
+    {
+        String tableName = "unpartitioned_absent_table_data";
+        String tablePath = "/user/hive/warehouse/" + tableName;
+
+        assertThat(query(format("CREATE TABLE IF NOT EXISTS %s (col INT)", tableName))).containsOnly(row(0));
```

Comment:
remove `IF NOT EXISTS`
if you want the test to be repeatable (even if fails), add `"DROP TABLE " + tableName` before the `CREATE`.
This ensures this is _our_ test table, with expected data & schema, not something else.